### DISCUSSION TITLE
how do i make to my messages history save and appear too, not just the messages from the AI?

### DISCUSSION THREAD

**Discussion by flimblet** [srcnalt/OpenAI-Unity#25] [Q&A] [answered]
i want to make my messages history be saved (so the AI can understand context) and to appear as well, not just the AI's. how do i do that?

explanation for saving messages history: i want to make my messages history be saved in the server (that by now will be my pc)

also, how do i make to my messages also be sent pressing enter?

**Comment by srcnalt** [comment, marked as answer]
@gGaroto these are not questions in the scope of this package but your project.
You can save the message history in a database by sending the string containing the messages. And as for sending by pressing enter, you can use the Unity Input system.

**Reply by flimblet** [reply]
oh yeah, right, sorry! but thank you!
